failed to register classes with Kryo

Hi @ryan-williams @arahuja 

                          please find the error when am trying to run the spark-submit.

16/05/02 14:59:37 INFO BlockManager: BlockManager stopped
16/05/02 14:59:37 INFO BlockManagerMaster: BlockManagerMaster stopped
16/05/02 14:59:37 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
16/05/02 14:59:37 INFO SparkContext: Successfully stopped SparkContext
**Exception in thread "main" org.apache.spark.SparkException: Failed to register classes with Kryo**
        at org.apache.spark.serializer.KryoSerializer.newKryo(KryoSerializer.scala:128)
        at org.apache.spark.serializer.KryoSerializerInstance.borrowKryo(KryoSerializer.scala:273)
        at org.apache.spark.serializer.KryoSerializerInstance.<init>(KryoSerializer.scala:258)
        at org.apache.spark.serializer.KryoSerializer.newInstance(KryoSerializer.scala:174)
        at org.apache.spark.broadcast.TorrentBroadcast$.blockifyObject(TorrentBroadcast.scala:201)
        at org.apache.spark.broadcast.TorrentBroadcast.writeBlocks(TorrentBroadcast.scala:102)
        at org.apache.spark.broadcast.TorrentBroadcast.<init>(TorrentBroadcast.scala:85)
        at org.apache.spark.broadcast.TorrentBroadcastFactory.newBroadcast(TorrentBroadcastFactory.scala:34)
        at org.apache.spark.broadcast.BroadcastManager.newBroadcast(BroadcastManager.scala:63)
        at org.apache.spark.SparkContext.broadcast(SparkContext.scala:1326)
        at org.hammerlab.guacamole.reference.ReferenceBroadcast$.readFasta(ReferenceBroadcast.scala:73)
        at org.hammerlab.guacamole.reference.ReferenceBroadcast$.apply(ReferenceBroadcast.scala:154)
        at org.hammerlab.guacamole.commands.jointcaller.SomaticJoint$Caller$.run(SomaticJointCaller.scala:92)
        at org.hammerlab.guacamole.commands.jointcaller.SomaticJoint$Caller$.run(SomaticJointCaller.scala:80)
        at org.hammerlab.guacamole.commands.SparkCommand.run(SparkCommand.scala:10)
        at org.hammerlab.guacamole.commands.Command.run(Command.scala:46)
        at org.hammerlab.guacamole.Main$.main(Main.scala:68)
        at org.hammerlab.guacamole.Main.main(Main.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:731)
        at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:181)
        at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:206)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:121)
        at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
**Caused by: java.lang.ClassNotFoundException: org.hammerlab.guacamole.kryo.GuacamoleKryoRegistrar**


Hi could you please let me know if there is any way to filter the variants based on quality or is  there any way to get the qual filter and other missing information in the vcf files.

Thanks & Regards,
Ankush Reddy.

--- a/src/main/scala/org/hammerlab/guacamole/commands/SparkCommand.scala
+++ b/src/main/scala/org/hammerlab/guacamole/commands/SparkCommand.scala
@@ -41,7 +41,7 @@ abstract class SparkCommand[T <% Args4jBase: Manifest] extends Command[T] {
     }
 
     if (config.getOption("spark.kryo.registrator").isEmpty) {
-      config.set("spark.kryo.registrator", "org.hammerlab.guacamole.kryo.GuacamoleKryoRegistrar")
+      config.set("spark.kryo.registrator", "org.hammerlab.guacamole.kryo.Registrar")
     }
 
     if (config.getOption("spark.kryoserializer.buffer").isEmpty) {

